tests/kafkatest/tests/streams/streams_upgrade_test.py

```diff
@@ -189,54 +189,6 @@ def test_upgrade_downgrade_brokers(self, from_version, to_version):
         processor.stop()
         processor.node.account.ssh_capture("grep SMOKE-TEST-CLIENT-CLOSED %s" % processor.STDOUT_FILE, allow_fail=False)
 
-    @matrix(from_version=metadata_2_versions, to_version=metadata_2_versions)
-    def test_simple_upgrade_downgrade(self, from_version, to_version):
```

Comment:
Moved this case to the new test file, `streams_application_upgrade_test.py` to avoid messing too much with the other tests here.

Reply:
Should we merge the `test_metadata_upgrade` below to the new file? I think with the new principle it is less meaningful to test the upgrade path from an old v1 to another old v2, since, the test of v2 should already cover that.

Reply:
Aha! For that, I've been preparing a separate PR: https://github.com/apache/kafka/pull/8971